### DISCUSSION TITLE
Is there a solution to completely disable udp traffic?

### DISCUSSION THREAD

**Discussion by chinesejp** [rustdesk/rustdesk#9967] [General]
Hi,
Rustdesk requires udp traffic to establish a connection by default, but our network environment does not allow udp traffic. When I disabled port 21116, Rustdesk could not fall back to using tcp to establish a connection, so the connection failed.
Is there a solution to completely disable udp traffic?
Thank you.

**Comment by 21pages** [comment]
https://rustdesk.com/docs/en/self-host/client-configuration/advanced-settings/#disable-udp

**Reply by mikekgr** [reply]
so, for example to disable
file transfer, shall I use in Windows os: 
%ProgramFiles%\RustDesk\rustdesk.exe  enable-file-transfer=N
Is the above correct?

**Reply by chinesejp** [reply]
Thanks.
Does this mean I need to change the client?


Controls whether to use TCP only. It will not use UDP 21116 any more, TCP 21116 will be used instead.
Values ​​Default Example
Y,N         N        disable-udp=Y

**Reply by 21pages** [reply]
Pro required.
![1732001460056](https://github.com/user-attachments/assets/d5e769e5-3da0-4e55-ad02-477d23e72058)

**Reply by chinesejp** [reply]
I don't think so.
[Docs of RustDesk](https://rustdesk.com/docs/en/) > [Self-host](https://rustdesk.com/docs/en/self-host/) > [Client Configuration](https://rustdesk.com/docs/en/self-host/client-configuration/) > Advanced Settings

**Reply by xlionjuan** [reply]
~~Modify `RustDesk2.toml`~~ But not work

**Comment by mitsukuri** [comment]
So how do we disable UDP in the end of the day? Or must we all buy a license to find out?
